python_modules/dagster/dagster/_core/remote_representation/code_location.py

```diff
@@ -767,9 +769,11 @@ def __init__(
     def _job_ref_to_snap(self, job_ref: "JobRefSnap") -> "JobDataSnap":
         from dagster._core.remote_representation.external_data import JobDataSnap
 
-        snapshot = self._job_snaps[job_ref.snapshot_id]
+        snapshot = self._job_snaps_by_name[job_ref.name]
         parent_snapshot = (
-            self._job_snaps[job_ref.parent_snapshot_id] if job_ref.parent_snapshot_id else None
+            self._job_snaps_by_snapshot_id.get(job_ref.parent_snapshot_id)
+            if job_ref.parent_snapshot_id
+            else None
```

Comment:
i am a little confused about when a parent snapshot would ever end up actually getting referenced here - isn't this only called for non-subset jobs anyway? since subsets require going out to the code server?